pkg/ottl/ottlfuncs/func_truncate.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TruncateArguments[K any] struct {
+	Target ottl.GetSetter[K] `ottlarg:"0"`
+	Limit  int64             `ottlarg:"1"`
+}
+
+func NewTruncateFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("truncate", &TruncateArguments[K]{}, createTruncateFunction[K])
+}
+
+func createTruncateFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*TruncateArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("TruncateFactory args must be of type *TruncateArguments[K]")
+	}
+
+	return truncate(args.Target, args.Limit)
+}
+
+func truncate[K any](target ottl.GetSetter[K], limit int64) (ottl.ExprFunc[K], error) {
```

Comment:
Can truncateAll be made to depend on this function?

Reply:
Ya the `pcommon.Value` vs `ottl.GetSetter` ruins this idea.  Thanks for humoring me, but lets revert it.  I'm good keeping the change in truncate_all to remove the extra limit check tho.